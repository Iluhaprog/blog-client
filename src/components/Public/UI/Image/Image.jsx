import styled from 'styled-components';
import PropTypes from 'prop-types';

const Image = styled.figure`
  width: ${({width}) => width}px;
  height: ${({height}) => height}px;
  border-radius: ${({round}) => round ? '50%' : '0'};
  overflow: hidden;
  
  img {
    width: 100%;
    height: 100%;
    object-fit: cover;
    object-position: center;
  }
`;

Image.propTypes = {
  width: PropTypes.number,
  height: PropTypes.number,
  round: PropTypes.bool,
};

export {Image};
