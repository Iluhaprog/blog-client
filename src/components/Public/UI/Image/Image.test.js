import {render} from '@testing-library/react';
import {Image} from './Image';

test('Should render Image component', () => {
  const {asFragment} = render(
    <Image
      width={20}
      height={20}
      round
    >
      <img src={''} alt={'alt'}/>
    </Image>
  );
  expect(asFragment()).toMatchSnapshot();
});
