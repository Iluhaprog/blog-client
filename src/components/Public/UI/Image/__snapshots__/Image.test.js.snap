// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Image component 1`] = `
<DocumentFragment>
  <figure
    class="sc-bdfBwQ doyRYB"
    height="20"
    width="20"
  >
    <img
      alt="alt"
      src=""
    />
  </figure>
</DocumentFragment>
`;
